src/Xamarin.Android.Build.Tasks/Tasks/Aapt2LinkAssetPack.cs

```diff
@@ -0,0 +1,75 @@
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Xml;
+using System.Xml.Linq;
+using Microsoft.Build.Utilities;
+using Microsoft.Build.Framework;
+using System.Text.RegularExpressions;
+using System.Collections.Generic;
+using Xamarin.Android.Tools;
+using Microsoft.Android.Build.Tasks;
+
+namespace Xamarin.Android.Tasks {
+
+	public class Aapt2LinkAssetPack : Aapt2 {
+		public override string TaskPrefix => "A2LAP";
+
+		[Required]
+		public ITaskItem Manifest { get; set; }
+
+		[Required]
+		public ITaskItem[] AssetDirectories { get; set; }
+
+		[Required]
+		public string PackageName { get; set; }
+
+		[Required]
+		public ITaskItem OutputArchive { get; set; }
+
+		public string OutputFormat { get; set; } = "proto";
+
+		protected override int GetRequiredDaemonInstances ()
+		{
+			return Math.Min (1, DaemonMaxInstanceCount);
+		}
+
+		public async override System.Threading.Tasks.Task RunTaskAsync ()
+		{
+			RunAapt (GenerateCommandLineCommands (Manifest, OutputArchive), OutputArchive.ItemSpec);
+			ProcessOutput ();
+			if (OutputFormat == "proto" && File.Exists (OutputArchive.ItemSpec)) {
+				// move the manifest to the right place.
+				using (var zip = new ZipArchiveEx (OutputArchive.ItemSpec, File.Exists (OutputArchive.ItemSpec) ? FileMode.Open : FileMode.Create)) {
+					zip.MoveEntry ("AndroidManifest.xml", "manifest/AndroidManifest.xml");
+					zip.Archive.DeleteEntry ("resources.pb");
```

Comment:
I was looking at `MoveEntry()` and how it has to extra into a `MemoryStream` and delete/add a new entry.

Is there a way to move the manifest in the task that creates the file?

Reply:
`aapt2` adds the manifest there isn't anything we can do about its location until after it has been created.